extensions/database-direct-access/src/main/java/de/sovity/edc/extension/db/directaccess/DirectDatabaseAccess.java

```diff
@@ -0,0 +1,35 @@
+/*
+ *  Copyright (c) 2024 sovity GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       sovity GmbH - initial API and implementation
+ *
+ */
+
+package de.sovity.edc.extension.db.directaccess;
+
+import lombok.RequiredArgsConstructor;
+import org.eclipse.edc.runtime.metamodel.annotation.ExtensionPoint;
+import org.jooq.DSLContext;
+import org.jooq.SQLDialect;
+import org.jooq.impl.DSL;
+
+import java.util.function.Supplier;
+import javax.sql.DataSource;
+
+@ExtensionPoint
+@RequiredArgsConstructor
+public class DirectDatabaseAccess {
+
+    private final Supplier<DataSource> dataSource;
```

Comment:
there is only one instance of the DataSource, so why do we need a supplier?

a DataSource is literally a Connection Factory